### PR TITLE
Update tree item height correctly on DPI change

### PR DESCRIPTION
**Pull request by akoch-yatta**
With the current implementation of the handler on DPI zoom changes, not all rescaling cases are covered correctly. If the item height was set at least once programmatically (currently done in Tree::setItemHeight), it will never be done by the OS anymore. With the current implementation a double scale up can happen (once scaled up from the OS and once from the handler), see:

image

To make it failsafe a new internal flag is used to evaluate whether the item height was set programmatically or not.

## How to reproduce

1. Adapt the VM argument of e.g. the Eclipse IDE by adding
```
-Dswt.autoScale=quarter
-Dswt.autoScale.updateOnRuntime=true
```
2. Have two monitors with different zoom, e.g. 100% and 150%
3. Start the IDE on 100%
4. Open the Outline view
5. Move it to the monitor with 150%  -> The item height will be too big as in the screenshot

### CODE REVIEW

File: bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Tree.java [hunk 1]
```diff
@@ -4347,7 +4348,11 @@ void setItemCount (int count, long hParent) {
 /*public*/ void setItemHeight (int itemHeight) {
 	checkWidget ();
 	if (itemHeight < -1) error (SWT.ERROR_INVALID_ARGUMENT);
-	OS.SendMessage (handle, OS.TVM_SETITEMHEIGHT, itemHeight, 0);
+	long newHeight = OS.SendMessage (handle, OS.TVM_SETITEMHEIGHT, itemHeight, 0);
```
Comment (fedejeanne) [inline bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Tree.java:4351]: According to the [documentation of TVM_SETITEMHEIGHT](https://learn.microsoft.com/en-us/windows/win32/controls/tvm-setitemheight#return-value):
> # Return value
> Returns the previous height of the items, in pixels.

So this one should be renamed to `oldHeight`.

File: bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Tree.java [hunk 2]
```diff
@@ -4347,7 +4348,11 @@ void setItemCount (int count, long hParent) {
...
+	// if the item height was set at least once programmatically, it will never be
+	// done by the OS anymore. To handle resizing after DPI changes, this must be
+	// tracked with the itemHeightSetProgrammatically-flag
```
Comment (fedejeanne) [inline bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Tree.java:4354]: It's kind of confusing what this comment wants to say, can you rephrase it?

If I interpret the changes correctly, the `if` in line 8288 can only work if the method `setItemHeight(int)` has been called at least twice, right?

Comment (akoch-yatta) [reply bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Tree.java:4354]: Sure, I will rephrase it a bit

But why twice?

File: bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Tree.java (hunk 1 above)
Comment (akoch-yatta) [reply bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Tree.java]: I updated the branch. The solution is now much simpler as before. The behavior can just be reset by setting it to -1.
